### ISSUE TITLE
Unsupported gpu architecture 'compute_70' (mac os)

### ISSUE THREAD

**Issue by twidlr** [fireice-uk/xmr-stak#492]
Hello,

I'm trying to compile on Mac Os High Sierra 10.13.2 (17C88)
GPU: GTX 980 ti

Everything checks out except for Building NVCC

I used:

```
make install
```

and get:

```
nvcc fatal   : Unsupported gpu architecture 'compute_70'
CMake Error at xmrstak_cuda_backend_generated_cuda_extra.cu.o.Release.cmake:219 (message):
  Error generating
/CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/nvcc_code/./xmrstak_cuda_backend_generated_cuda_extra.cu.o
```

Mining works but does only run on the CPUs

Does anyone know how to resolve this?

Thanks

**Comment by drewwiley** [comment]
Same issue here except compute_20 is not supported. Or sometimes the output will say make reference to libdevice sm_20,  I think.

**Comment by psychocrypt** [comment]
Cuda 8 is not supporting 70 therefore the issue. @drew has the oposit issue
in an other issue. He is using cuda9 where 20 is not supported.
I am not sure but it looks like there is an osx cmake issue in the version
check.
Run ccmake and remove the 70 from CUDA_ARCH should solve your issue.

**Comment by twidlr** [comment]
Thank you for your reply @psychocrypt

I checked my CUDA version, I have Cuda 9.0 installed

I will try you suggestions, and report if it works.

Cheers

**Comment by bmwbmerz8** [comment]
Trying adding -DCUDA_COMPILER=clang to the cmake, this solved this issue for me.

**Comment by esetnik** [comment]
@bmwbmerz8 using `-DCUDA_COMPILER=clang`:

```
[ 86%] Building CXX object CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/nvcc_code/cuda_core.cu.o
clang: error: cannot find libdevice for sm_20. Provide path to different CUDA installation via --cuda-path, or pass -nocudalib to build without linking with libdevice.
make[2]: *** [CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/nvcc_code/cuda_core.cu.o] Error 1
make[1]: *** [CMakeFiles/xmrstak_cuda_backend.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by psychocrypt** [comment]
please empty the build directory after you switched from cuda 9 to cuda8.

**Comment by psychocrypt** [comment]
This bug should be solved with #532. Please report back if this fix is working for your case.
If you not familiar with git than you can download the branch for testing from https://github.com/psychocrypt/xmr-stak/archive/fix-CUDA9andOSCCompile.zip

**Comment by esetnik** [comment]
@psychocrypt I tested your branch and am receiving the same problem using clang `cmake . -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl -DOpenCL_ENABLE=OFF -DCUDA_COMPILER=clang`

```
❯ make install
Scanning dependencies of target xmr-stak-c
[  3%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_blake256.c.o
[  6%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_groestl.c.o
[ 10%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_jh.c.o
[ 13%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_keccak.c.o
[ 16%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_skein.c.o
[ 20%] Linking C static library bin/libxmr-stak-c.a
[ 20%] Built target xmr-stak-c
Scanning dependencies of target xmr-stak-backend
[ 23%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/jconf.cpp.o
[ 26%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/version.cpp.o
[ 30%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/cpu/hwlocMemory.cpp.o
[ 33%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/cpu/jconf.cpp.o
[ 36%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/cpu/minethd.cpp.o
[ 40%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/backendConnector.cpp.o
[ 43%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/globalStates.cpp.o
[ 46%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/cpu/crypto/cryptonight_common.cpp.o
[ 50%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/http/httpd.cpp.o
[ 53%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/http/webdesign.cpp.o
[ 56%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/misc/console.cpp.o
[ 60%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/misc/executor.cpp.o
[ 63%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/misc/telemetry.cpp.o
[ 66%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/misc/utility.cpp.o
[ 70%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/net/jpsock.cpp.o
[ 73%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/net/socket.cpp.o
[ 76%] Linking CXX static library bin/libxmr-stak-backend.a
[ 76%] Built target xmr-stak-backend
Scanning dependencies of target xmr-stak
[ 80%] Building CXX object CMakeFiles/xmr-stak.dir/xmrstak/cli/cli-miner.cpp.o
[ 83%] Linking CXX executable bin/xmr-stak
[ 83%] Built target xmr-stak
Scanning dependencies of target xmrstak_cuda_backend
[ 86%] Building CXX object CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/nvcc_code/cuda_core.cu.o
clang: error: cannot find libdevice for sm_61. Provide path to different CUDA installation via --cuda-path, or pass -nocudalib to build without linking with libdevice.
make[2]: *** [CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/nvcc_code/cuda_core.cu.o] Error 1
make[1]: *** [CMakeFiles/xmrstak_cuda_backend.dir/all] Error 2
make: *** [all] Error 2
```

```
❯ cmake -LA
-- Set miner currency to 'monero' and 'aeon'
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/ethan/Downloads/xmr-stak-fix-CUDA9andOSCCompile
-- Cache values
CMAKE_AR:FILEPATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ar
CMAKE_BUILD_TYPE:STRING=Release
CMAKE_COLOR_MAKEFILE:BOOL=ON
CMAKE_CXX_COMPILER:FILEPATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
CMAKE_CXX_FLAGS:STRING=
CMAKE_CXX_FLAGS_DEBUG:STRING=-g
CMAKE_CXX_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG
CMAKE_CXX_FLAGS_RELEASE:STRING=-O3 -DNDEBUG
CMAKE_CXX_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
CMAKE_C_COMPILER:FILEPATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
CMAKE_C_FLAGS:STRING=
CMAKE_C_FLAGS_DEBUG:STRING=-g
CMAKE_C_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG
CMAKE_C_FLAGS_RELEASE:STRING=-O3 -DNDEBUG
CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
CMAKE_EXE_LINKER_FLAGS:STRING=
CMAKE_EXE_LINKER_FLAGS_DEBUG:STRING=
CMAKE_EXE_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_EXE_LINKER_FLAGS_RELEASE:STRING=
CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_EXPORT_COMPILE_COMMANDS:BOOL=OFF
CMAKE_INSTALL_NAME_TOOL:FILEPATH=/usr/bin/install_name_tool
CMAKE_INSTALL_PREFIX:PATH=/Users/ethan/Downloads/xmr-stak-fix-CUDA9andOSCCompile
CMAKE_LINKER:FILEPATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
CMAKE_LINK_STATIC:BOOL=OFF
CMAKE_MAKE_PROGRAM:FILEPATH=/usr/bin/make
CMAKE_MODULE_LINKER_FLAGS:STRING=
CMAKE_MODULE_LINKER_FLAGS_DEBUG:STRING=
CMAKE_MODULE_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_MODULE_LINKER_FLAGS_RELEASE:STRING=
CMAKE_MODULE_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_NM:FILEPATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/nm
CMAKE_OBJCOPY:FILEPATH=CMAKE_OBJCOPY-NOTFOUND
CMAKE_OBJDUMP:FILEPATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/objdump
CMAKE_OSX_ARCHITECTURES:STRING=
CMAKE_OSX_DEPLOYMENT_TARGET:STRING=
CMAKE_OSX_SYSROOT:STRING=
CMAKE_RANLIB:FILEPATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib
CMAKE_SHARED_LINKER_FLAGS:STRING=
CMAKE_SHARED_LINKER_FLAGS_DEBUG:STRING=
CMAKE_SHARED_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_SHARED_LINKER_FLAGS_RELEASE:STRING=
CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_SKIP_INSTALL_RPATH:BOOL=NO
CMAKE_SKIP_RPATH:BOOL=NO
CMAKE_STATIC_LINKER_FLAGS:STRING=
CMAKE_STATIC_LINKER_FLAGS_DEBUG:STRING=
CMAKE_STATIC_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_STATIC_LINKER_FLAGS_RELEASE:STRING=
CMAKE_STATIC_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_STRIP:FILEPATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip
CMAKE_VERBOSE_MAKEFILE:BOOL=FALSE
CPU_ENABLE:BOOL=ON
CUDA_64_BIT_DEVICE_CODE:BOOL=ON
CUDA_ARCH:STRING=61
CUDA_ATTACH_VS_BUILD_RULE_TO_CUDA_FILE:BOOL=ON
CUDA_BUILD_CUBIN:BOOL=OFF
CUDA_BUILD_EMULATION:BOOL=OFF
CUDA_COMPILER:STRING=clang
CUDA_CUDART_LIBRARY:FILEPATH=/usr/local/cuda/lib/libcudart.dylib
CUDA_CUDA_LIBRARY:FILEPATH=/usr/local/cuda/lib/libcuda.dylib
CUDA_ENABLE:BOOL=ON
CUDA_GENERATED_OUTPUT_DIR:PATH=
CUDA_HOST_COMPILATION_CPP:BOOL=ON
CUDA_HOST_COMPILER:FILEPATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
CUDA_KEEP_FILES:BOOL=OFF
CUDA_NVCC_EXECUTABLE:FILEPATH=/usr/local/cuda/bin/nvcc
CUDA_NVCC_FLAGS:STRING=
CUDA_NVCC_FLAGS_DEBUG:STRING=
CUDA_NVCC_FLAGS_MINSIZEREL:STRING=
CUDA_NVCC_FLAGS_RELEASE:STRING=
CUDA_NVCC_FLAGS_RELWITHDEBINFO:STRING=
CUDA_PROPAGATE_HOST_FLAGS:BOOL=ON
CUDA_SDK_ROOT_DIR:PATH=CUDA_SDK_ROOT_DIR-NOTFOUND
CUDA_SEPARABLE_COMPILATION:BOOL=OFF
CUDA_SHOW_REGISTER:BOOL=OFF
CUDA_TOOLKIT_INCLUDE:PATH=/usr/local/cuda/include
CUDA_TOOLKIT_ROOT_DIR:PATH=/usr/local/cuda
CUDA_USE_STATIC_CUDA_RUNTIME:BOOL=ON
CUDA_VERBOSE_BUILD:BOOL=OFF
CUDA_VERSION:STRING=9.0
CUDA_cublas_LIBRARY:FILEPATH=/usr/local/cuda/lib/libcublas.dylib
CUDA_cublas_device_LIBRARY:FILEPATH=/usr/local/cuda/lib/libcublas_device.a
CUDA_cudadevrt_LIBRARY:FILEPATH=/usr/local/cuda/lib/libcudadevrt.a
CUDA_cudart_static_LIBRARY:FILEPATH=/usr/local/cuda/lib/libcudart_static.a
CUDA_cufft_LIBRARY:FILEPATH=/usr/local/cuda/lib/libcufft.dylib
CUDA_cupti_LIBRARY:FILEPATH=/usr/local/cuda/extras/CUPTI/lib/libcupti.dylib
CUDA_curand_LIBRARY:FILEPATH=/usr/local/cuda/lib/libcurand.dylib
CUDA_cusolver_LIBRARY:FILEPATH=/usr/local/cuda/lib/libcusolver.dylib
CUDA_cusparse_LIBRARY:FILEPATH=/usr/local/cuda/lib/libcusparse.dylib
CUDA_nppc_LIBRARY:FILEPATH=/usr/local/cuda/lib/libnppc.dylib
CUDA_nppial_LIBRARY:FILEPATH=/usr/local/cuda/lib/libnppial.dylib
CUDA_nppicc_LIBRARY:FILEPATH=/usr/local/cuda/lib/libnppicc.dylib
CUDA_nppicom_LIBRARY:FILEPATH=/usr/local/cuda/lib/libnppicom.dylib
CUDA_nppidei_LIBRARY:FILEPATH=/usr/local/cuda/lib/libnppidei.dylib
CUDA_nppif_LIBRARY:FILEPATH=/usr/local/cuda/lib/libnppif.dylib
CUDA_nppig_LIBRARY:FILEPATH=/usr/local/cuda/lib/libnppig.dylib
CUDA_nppim_LIBRARY:FILEPATH=/usr/local/cuda/lib/libnppim.dylib
CUDA_nppist_LIBRARY:FILEPATH=/usr/local/cuda/lib/libnppist.dylib
CUDA_nppisu_LIBRARY:FILEPATH=/usr/local/cuda/lib/libnppisu.dylib
CUDA_nppitc_LIBRARY:FILEPATH=/usr/local/cuda/lib/libnppitc.dylib
CUDA_npps_LIBRARY:FILEPATH=/usr/local/cuda/lib/libnpps.dylib
HWLOC:FILEPATH=/usr/local/lib/libhwloc.dylib
HWLOC_ENABLE:BOOL=ON
HWLOC_INCLUDE_DIR:PATH=/usr/local/include
MHTD:FILEPATH=/usr/local/lib/libmicrohttpd.dylib
MICROHTTPD_ENABLE:BOOL=ON
MTHD_INCLUDE_DIR:PATH=/usr/local/include
OPENSSL_CRYPTO_LIBRARY:FILEPATH=/usr/local/opt/openssl/lib/libcrypto.dylib
OPENSSL_INCLUDE_DIR:PATH=/usr/local/opt/openssl/include
OPENSSL_SSL_LIBRARY:FILEPATH=/usr/local/opt/openssl/lib/libssl.dylib
OpenCL_ENABLE:BOOL=OFF
OpenSSL_ENABLE:BOOL=ON
PKG_CONFIG_EXECUTABLE:FILEPATH=/usr/local/bin/pkg-config
XMR-STAK_COMPILE:STRING=native
XMR-STAK_CURRENCY:STRING=all
XMR-STAK_LARGEGRID:BOOL=ON
XMR-STAK_THREADS:STRING=0
```
